[GTK][WPE] Add webkit_web_hit_test_result_get_js_node() and deprecate the rest of the DOM APIs

carlosgcampos opened the pull request:
#### b51fb227ee726aa93c621ad5d185a2ed1ee24dfa
<pre>
[GTK][WPE] Add webkit_web_hit_test_result_get_js_node() and deprecate the rest of the DOM APIs
<a href="https://bugs.webkit.org/show_bug.cgi?id=249020">https://bugs.webkit.org/show_bug.cgi?id=249020</a>

Reviewed by Michael Catanzaro.

* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/WebKitDOMNode.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/WebKitDOMObject.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp:
(webkit_frame_get_js_value_for_dom_object):
(webkit_frame_get_js_value_for_dom_object_in_script_world):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.cpp:
(webkitWebHitTestResultGetProperty):
(webkitWebHitTestResultSetProperty):
(webkit_web_hit_test_result_class_init):
(webkitWebHitTestResultCreate):
(webkit_web_hit_test_result_get_node):
(webkit_web_hit_test_result_get_js_node):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMNode.h:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMObject.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMNode.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMObject.h:
* Tools/TestWebKitAPI/Tests/WebKitGLib/FrameTest.cpp:
(WebKitFrameTest::testJavaScriptValues):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp:
(contextMenuCallback):

Canonical link: <a href="https://commits.webkit.org/257783@main">https://commits.webkit.org/257783@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fa426b062dec70d1d1ee74d28eafea0959c9026b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100010 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9180 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33090 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109356 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/169591 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104009 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10061 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86508 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92456 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107256 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105778 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/7601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90886 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34318 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/22279 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2973 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/23794 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2933 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/46163 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9066 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43272 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4782 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2747 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.cpp:
```diff
@@ -148,10 +178,51 @@ WebKitWebHitTestResult* webkitWebHitTestResultCreate(const HitTestResult& hitTes
+/**
+ * webkit_web_hit_test_result_get_js_node:
+ * @hit_test_result: a #WebKitWebHitTestResult
+ * @world: (nullable): a #WebKitScriptWorld
```
Should mention that %NULL uses the default script world

mcatanzaro reviewed (approved):
The changes look good here, but the title is too misleading. How about:

"Add webkit_web_hit_test_result_get_js_node() and deprecate the rest of the DOM APIs"

which is currently what you have written in the description line. Just try to make the title reflect the extent of the change better. (A bit tough when you have so much in one pull request, but it would be awkward to split this into multiple pull requests, so just changing the title should suffice IMO.)

carlosgcampos committed:
[GTK][WPE] Add webkit_web_hit_test_result_get_js_node() and deprecate the rest of the DOM APIs
https://bugs.webkit.org/show_bug.cgi?id=249020

Reviewed by Michael Catanzaro.

* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/WebKitDOMNode.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/DOM/WebKitDOMObject.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp:
(webkit_frame_get_js_value_for_dom_object):
(webkit_frame_get_js_value_for_dom_object_in_script_world):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.cpp:
(webkitWebHitTestResultGetProperty):
(webkitWebHitTestResultSetProperty):
(webkit_web_hit_test_result_class_init):
(webkitWebHitTestResultCreate):
(webkit_web_hit_test_result_get_node):
(webkit_web_hit_test_result_get_js_node):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebHitTestResult.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMNode.h:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMObject.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMNode.h:
* Source/WebKit/WebProcess/InjectedBundle/API/wpe/DOM/WebKitDOMObject.h:
* Tools/TestWebKitAPI/Tests/WebKitGLib/FrameTest.cpp:
(WebKitFrameTest::testJavaScriptValues):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebExtensionTest.cpp:
(contextMenuCallback):

Canonical link: https://commits.webkit.org/257783@main
